File: bankingtool.client/src/app/dto/request/user/save-user-request.ts
import { RequestId } from "../request-id";

export interface SaveUserRequestObject {
  request: SaveUserRequest;
}

export interface SaveUserRequest {
  user: SaveUserRequestUser;
  state: SaveUserRequestState;
  city: SaveUserRequestCity;
  role: SaveUserRequestRole;
}

export interface SaveUserRequestUser {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
}

export interface SaveUserRequestState extends RequestId {

}

export interface SaveUserRequestCity extends RequestId {

}

export interface SaveUserRequestRole extends RequestId {

}

File: BankingTool.Model/Dto/RequestDto/SaveUserRequest.cs
namespace BankingTool.Model.Dto.RequestDtos
{
    public class SaveUserRequestObject
    {
        public SaveUserRequest Request { get; set; }
    }
    public class SaveUserRequest
    {
        public SaveUserRequestUser User { get; set; }
        public SaveUserRequestState State { get; set; }
        public SaveUserRequestCity City { get; set; }
        public SaveUserRequestRole Role { get; set; }
    }
    public class SaveUserRequestUser
    {
        public string Email { get; set; }
        public string Password { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
    }
    public class SaveUserRequestState : RequestId
    {
    }
    public class SaveUserRequestCity : RequestId
    {
    }
    public class SaveUserRequestRole : RequestId
    {
    }
}
